### PR TITLE
Write to output when no casm_output_path provided

### PR DESCRIPTION
**Pull request by war-in**

### CODE REVIEW

**Review by piotmag769 — COMMENTED** [review]
I think it should stay as required, I don't really see a use case for that (simple debugging is easier when you have the sierra in the file too imo)

**Comment by Arcticae** [comment]
@piotmag769 if you want to omit writing file to the hard drive it's really useful since you would just call the binary in other process and read from stdout instead of performing read and write from disk (which is much more expensive)

**Review by Arcticae — APPROVED** [review]

**Review by Arcticae — CHANGES_REQUESTED** [review]
Actually we should e2e test this somehow (match the output against a regex?)

**Review by Arcticae — APPROVED** [review]
Ignore my last review lol i didn't notice the test

File: src/main.rs [hunk 1]
```diff
@@ -34,14 +34,21 @@ fn main_execution() -> Result<bool> {
...
-        .create(true)
-        .append(false)
-        .open(args.casm_output_path)
-        .context("Unable to open/create casm json file")?;
+    match args.casm_output_path {
+        Some(output_path) => {
+            let casm_file = OpenOptions::new()
+                .write(true)
+                .create(true)
+                .append(false)
```
Comment (piotmag769) [inline src/main.rs:42]: I think we need `truncate(true)` too
https://stackoverflow.com/questions/65782872/how-to-delete-the-contents-of-a-file-before-writing-to-it

File: tests/e2e.rs [hunk 2]
```diff
@@ -58,6 +58,20 @@ fn write_to_existing_file() {
...
+    let sierra_file_name = "sierra_1_4_0.json";
+    let args = vec!["--sierra-input-path", &sierra_file_name];
+
+    let temp_dir = prepare_temp_dir(sierra_file_name);
+    let snapbox = prepare_runner(args, &temp_dir);
+
+    let output = String::from_utf8(snapbox.assert().success().get_output().stdout.clone()).unwrap();
+    assert!(output.contains("bytecode"));
+
+    fs::remove_dir_all(temp_dir).unwrap();
```
Comment (piotmag769) [inline tests/e2e.rs:72]: Unnecessary

File: tests/e2e.rs [hunk 3]
```diff
@@ -58,6 +58,20 @@ fn write_to_existing_file() {
     fs::remove_dir_all(temp_dir).unwrap();
 }
 
+#[test]
+fn write_to_stdout() {
+    let sierra_file_name = "sierra_1_4_0.json";
+    let args = vec!["--sierra-input-path", &sierra_file_name];
+
+    let temp_dir = prepare_temp_dir(sierra_file_name);
+    let snapbox = prepare_runner(args, &temp_dir);
```
Comment (piotmag769) [inline tests/e2e.rs:67]: I think it should be rebased, the names changed

**Pull request merged**